src/DocumentDB.ChangeFeedProcessor/ChangeFeedProcessorBuilder.cs

```diff
@@ -396,12 +396,13 @@ private async Task<IPartitionManager> BuildPartitionManagerAsync(ILeaseManager l
                 this.partitionProcessorFactory ?? new PartitionProcessorFactory(this.feedDocumentClient, this.changeFeedProcessorOptions, leaseManager, collectionSelfLink),
                 this.changeFeedProcessorOptions);
 
-            var partitionController = new PartitionController(leaseManager, partitionObserverFactory, synchronizer);
             if (this.loadBalancingStrategy == null)
             {
                 this.loadBalancingStrategy = new EqualPartitionsBalancingStrategy(this.HostName, this.changeFeedProcessorOptions.MinPartitionCount, this.changeFeedProcessorOptions.MaxPartitionCount, this.changeFeedProcessorOptions.LeaseExpirationInterval);
             }
 
+            IPartitionController partitionController = new PartitionController(leaseManager, partitionObserverFactory, synchronizer);
+            partitionController = new PartitionControllerHealthnessEvaluator(partitionController, this.changeFeedProcessorOptions.LeaseExpirationInterval, new FailFastUnhealthyHandlingStrategy());
```

Comment:
nit: why reassign the variable? Just to avoid creating a new one? It would be more clear IMHO like this:

    IPartitionController partitionController = new PartitionControllerHealthnessEvaluator(
        new PartitionController(leaseManager, partitionObserverFactory, synchronizer), 
        this.changeFeedProcessorOptions.LeaseExpirationInterval, 
        new FailFastUnhealthyHandlingStrategy());

Reply:
we use this style to see all decorator levels.

ISomeInterface x = new SomeImplementation();
x = new Decorator1(x); // keeping same interface contract
x = new Decorator2(x);

so we see all decorating levels